src/conv/problem_description.cpp

```diff
@@ -176,6 +198,17 @@ void ProblemDescription::Serialize(std::ostream& stream) const
         // Group count > 1 identifies Group/Depthwise modes.
         if(GetGroupCount() != 1)
             optional << 'g' << GetGroupCount();
+
+        if(GetInCastType() || GetWeightsCastType() || GetOutCastType())
+        {
+            std::tuple<miopenDataType_t, std::optional<miopenDataType_t>> in_cast =
+                std::make_tuple(GetInDataType(), GetInCastType());
+            std::tuple<miopenDataType_t, std::optional<miopenDataType_t>> weights_cast =
+                std::make_tuple(GetWeightsDataType(), GetWeightsCastType());
+            std::tuple<miopenDataType_t, std::optional<miopenDataType_t>> out_cast =
+                std::make_tuple(GetOutDataType(), GetOutCastType());
+            optional << "c" << EncodeDataTypesForKey(in_cast, weights_cast, out_cast);
+        }
```

Comment:
[R] Save space in db key and optimize code. Do not print casting value when casting is not actually happens. Get rid of ~crutches~ tuples ;)
```suggestion
        if(const auto ct = GetInCastType())
            optional << "_ci" << GetDataTypeName(*ct);
        if(const auto ct = GetWeightsCastType())
            optional << "_cw" << GetDataTypeName(*ct);
        if(const auto ct = GetOutCastType())
            optional << "_co" << GetDataTypeName(*ct);
```

Reply:
This makes sense, we should minimize the changes we make to DBs